js/components/accordion.js

```diff
@@ -34,22 +60,90 @@
       var target = e.target || e.srcElement;
 
       var accordionItem = this.findParentNode('accordion-item', target),
-        accordionStatus = accordionItem.getAttribute('accordion-open');
+        contentSelector = '#' + target.getAttribute('aria-controls'),
+        contentElem = accordionItem.querySelector(contentSelector);
 
-      accordionStatus = (accordionStatus == 'true') ? 'false' : 'true';
-      accordionItem.setAttribute('accordion-open', accordionStatus);
+      this.toggleAttribute(accordionItem, 'aria-expanded');
+      this.toggleAttribute(contentElem, 'aria-hidden')
     },
-    /**
-     * Event handler that binds
-     * to the toggleAccordion function
-     *
-     * @return void
-     */
-    registerEventListeners: function () {
-      for (var i = 0; i < this.accordionButtons.length; i++) {
-        this.accordionButtons[i].addEventListener('click',
-          this.toggleAccordion.bind(this));
+
+    // sets [accordion-item], [accordion-button], [accordion-content]
+    setAttributes: function () {
+      for (var i = 0; i < this.accordions.length; i++) {
+
+        var accordion = this.accordions[i],
+          accordionName = accordion.getAttribute('accordion'),
+          accordionItems;
+
+        if (accordion.hasAttribute('accordion-item')) {
+          accordionItems = [accordion];
+        } else {
+          accordionItems = accordion.querySelectorAll('[accordion-item]');
+          // check if node list is different than list of children
+          if (accordionItems.length !== accordion.children.length) {
+            this.forEach(accordion.children, function(index, item) {
+              // add [accordion-item] attribute to list items
+              if (item.tagName.toLowerCase() === 'li' &&
+                !item.hasAttribute('accordion-item')) {
+                item.setAttribute('accordion-item', true);
+              }
+            });
```

Comment:
I agree that making loops within loops is generally bad practice, but definitely the most straightforward approach when I am dealing with lists of lists.